solutions/src/main/java/ru/sberbank/school/task02/FxConversionServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package ru.sberbank.school.task02;
+
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
+import ru.sberbank.school.task02.util.Symbol;
+import ru.sberbank.school.task02.exception.FxConversionException;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+
+
+public class FxConversionServiceImpl implements FxConversionService{
+
+    ExternalQuotesService externalQuotesService;
+
+    FxConversionServiceImpl(ExternalQuotesService externalQuotesService){
+        this.externalQuotesService = externalQuotesService;
+    }
+
+    @Override
+    public BigDecimal convert(ClientOperation operation, Symbol symbol, BigDecimal amount) throws FxConversionException {
+
+        List<Quote> quotes = new ArrayList<>();
+        quotes = externalQuotesService.getQuotes(symbol);
+
+        if(quotes.size() == 0)
+            throw new FxConversionException("Список котировок ExternalQuotesService пустой");
+        else {
+
+            //Поиск первого элемента, значение которого больше amount
+            int k = 0;
+            while (quotes.get(k).getVolumeSize().compareTo(amount) < 0 &&
+                    k < quotes.size() - 1) {
+                k++;
+            }
+
+            //Индекс минимального элемента, который больше amount
+            int minIndex = k;
+            for (int i = k; i < quotes.size(); i++) {
+
+                if (quotes.get(i).getVolumeSize().compareTo(amount) > 0) {
+                    if (quotes.get(i).getVolumeSize().compareTo(quotes.get(minIndex).getVolumeSize()) < 0) {
+                        minIndex = i;
+                    }
+                }
+            }
+
+
+            if (operation == ClientOperation.BUY) {
+                return quotes.get(minIndex).getOffer();
+            }
+            else if (operation == ClientOperation.SELL) {
+                return quotes.get(minIndex).getBid();
+            }
```

Comment:
&#39;}&#39; at column 13 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-566990509>*